src/types/uint.rs

```diff
@@ -115,13 +115,23 @@ macro_rules! impl_uint {
       fn from(mut num: u64) -> Self {
         let mut arr = [0; $len];
         for i in 0..8 {
-          arr[$len - 1 - i] =  num as u8;
+          arr[$len - 1 - i] = num as u8;
           num = num >> 8;
         }
         $name(arr)
       }
     }
 
+    impl From<$name> for u64 {
```

Comment:
This kind of implies that the conversion is lossless, I would rather have `low_u64()` as we have in bigint.